src/storage-area.cpp

```diff
@@ -0,0 +1,221 @@
+}
+
+OrthancPluginErrorCode StorageCreateCallback(const char *uuid,
+                                             const void *content,
+                                             int64_t size,
+                                             OrthancPluginContentType type) {
+    fs::path path;
+    switch (type) {
+        case OrthancPluginContentType_Dicom:
+            return WriteDicomFile(DicomFile(content, size), uuid);
+        case OrthancPluginContentType_DicomAsJson:
+        case OrthancPluginContentType_DicomUntilPixelData:
+        case _OrthancPluginContentType_INTERNAL:
+        case OrthancPluginContentType_Unknown:
+            path = GetPath(type,uuid);
+    }
+    std::fstream file(path, std::ios::out | std::ios::binary);
+    if (file.is_open()) {
+        file.write((const char *) content, size);
+        if (file.good()) {
+            file.close();
+            return OrthancPluginErrorCode_Success;
+        }
+        if(globals::context) OrthancPluginLogWarning(globals::context, "StorageCreateCallback: but write out appears bad");
+    }
+    return OrthancPluginErrorCode_FileStorageCannotWrite;
+}
+
+OrthancPluginErrorCode StorageReadWholeCallback(OrthancPluginMemoryBuffer64 *target,
+                                                const char *uuid,
+                                                OrthancPluginContentType type) {
+    auto path = GetPath(type,uuid);
+    // todo: is the buffer ready? without this call..
+    OrthancPluginCreateMemoryBuffer64(globals::context, target, fs::file_size(path));
+    std::fstream file(path, std::ios::in | std::ios::binary);
+    if(file.is_open()){
+        file.read((char*)target->data, target->size);
+        if(file.good()){
+            file.close();
+            return OrthancPluginErrorCode_Success;
+        }
+        if(globals::context) OrthancPluginLogWarning(globals::context, "StorageReadWholeCallback: opened file, but couldn't read it");
+        return OrthancPluginErrorCode_StorageAreaPlugin;
+    }
+    return OrthancPluginErrorCode_InexistentFile;
+}
+
+OrthancPluginErrorCode StorageReadRangeCallback(OrthancPluginMemoryBuffer64 *target,
+                                                const char *uuid,
+                                                OrthancPluginContentType type,
+                                                uint64_t rangeStart) {
+    auto path = GetPath(type,uuid);
+    // todo: is the buffer ready? without this call..
+    OrthancPluginCreateMemoryBuffer64(globals::context, target, fs::file_size(path));
+    std::fstream file(path, std::ios::in | std::ios::binary);
+    if(file.is_open()){
+        file.seekg(rangeStart);
+        file.read((char*)target->data, target->size);
+        return OrthancPluginErrorCode_Success;
+    }
+    return OrthancPluginErrorCode_InexistentFile;
+}
+
+OrthancPluginErrorCode StorageRemoveCallback(const char *uuid, OrthancPluginContentType type) {
+    switch(type){
```

Comment:
Other file types? Maybe use GetPath directly to retrieve file path

Reply:
GetPath doesn't really work for the other types